app/src/main/java/org/fossasia/phimpme/share/SharingActivity.java

```diff
@@ -378,6 +380,17 @@ public void onClick(DialogInterface dialogInterface, int i) {
         dialogBuilder.show();
     }
 
+    /**
+     * Method to send the result of the share operation
+     * @param code - SUCCESS if user shares it FAIL otherwise.
+     */
+    private void sendResult(int code) {
+        Intent result = new Intent();
+        result.putExtra(Constants.SHARE_RESULT, code);
+        setResult(RESULT_OK, result);
+        finish();
```

Comment:
This is finishing the share activity before the image is even shared. Just try to share an image to imgur.

Reply:
I need forgot to check for AsyncTasks. I made the corrections now.
Please review!
Thanks